tensorflow/core/kernels/conv_ops_gpu_3.cu.cc

```diff
@@ -439,17 +439,22 @@ struct GreaterThan {
   constexpr bool operator()(int a, int b) const { return a > b; }
 };
 
-// Tile size posibility frontier denotes the tile size combinations that consume
-// the most computational resources constrained by
+// The performance subspace refers to a subspace of the tile size space consists
+// of all tile sizes satisfying the constraint of
 // - number of threads per SM limit,
 // - shared memory limit and
 // - some experimentally determined, type-specific constraint on the product of
 // two side lengths to increase grid-level parallelism.
+// Tile size combinations lying outside this subspace are either not possible,
+// or are slower than the alternatives
 //
-// Tile size combinations lying on this frontier would achieve the maximum
-// utilization of available resources, and combinations lying outside this
-// frontier are either not possible, or are slower than the alternatives.
-
+// Tile size posibility frontier denotes the tile size combinations in that
+// consume the most computational resources given a particular long side len.
+//
+// It is worth noting that due to the discrete nature of tile size space, the
+// frontier does not constitute the boundary of the performant subspace. For
+// this reason, there are tile size combinations that lie on the boundary of
+// this performant subspace, but not on the frontier.
```

Comment:
AIUI the purpose of introducing the notion of the performant subspace is so that we can say "lies on the boundary of the subspace, but not the frontier."

But I think we can simplify this for readers.  What if we said something to the effect of what you had before about the frontier, and then defined the concept of "on the frontier in the short dim but not the long dim"?  (Or is it on the long dim but not the short dim?  Anyway.)

I think that would obviate a lot of the confusion I'm having reading this patch now, where I have to keep straight the "boundary" as separate from the "frontier", even though those words are synonymous in my mind, and even though the two divisions are closely related.

WDYT?

Reply:
I agree this way is much more economical.